From f8cf7e1aad42b52eb12b4e66aef59f876fe76cc8 Mon Sep 17 00:00:00 2001
From: Josef Schlehofer <pepe.schlehofer@gmail.com>
Date: Fri, 12 Feb 2021 16:15:09 +0100
Subject: [PATCH] setup.py: don't use exact version paho-mqtt

The thing is that in distribution (in this case OpenWrt / TurrisOS),
there is an issue that anyone with commit access can
backport/cherry-pick updates from master (dev) branch to stable branch.

Let's give an example:
reForis wants to use paho-mqtt with version 1.5.0, but in OpenWrt there
is version 1.4.0. However, compilation success and we will know it
once we try to run updated reForis that it shows 500 Internal Error and
by enabling lighttpd error log we can see:

pkg_resources.DistributionNotFound: The 'paho-mqtt==1.5.0' distribution was not found and is required by reforis

There are two solutions:
- remove pinning to specific versions (that's what we prefer)
This is similar to 714ffc7747b45e94e76d7791711de8d0c0a6756d.
- add pinned versions to minimal versions (e.g. >=)
This has its own advantages and disadvantages.
I mean that you were using some features from new versions of
dependencies and we will dont know it and reForis at least run, but smth
is broken and this has solution that you will let us know that we will
update it to some version in distribution
---
 setup.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/setup.py b/setup.py
index 72e38f47..54d107dd 100644
--- a/setup.py
+++ b/setup.py
@@ -47,7 +47,7 @@ setuptools.setup(
         'flup',
         'cachelib',
         'foris-client @ git+https://gitlab.nic.cz/turris/foris-controller/foris-client.git',
-        'paho-mqtt==1.5.0',
+        'paho-mqtt',
     ],
     setup_requires=[
         'Babel',
-- 
GitLab

